xbmc/settings/DisplaySettings.cpp

```diff
@@ -792,6 +792,26 @@ void CDisplaySettings::SettingOptionsMonitorsFiller(SettingConstPtr setting, std
     }
     list.push_back(std::make_pair(monitors[i], monitors[i]));
   }
+#elif defined(HAVE_WAYLAND)
```

Comment:
X11 behaves like this: assume you have 2 monitors connected. typical use case is a normal monitor or tv and a projector. users set monitor to projector. if you switch off the projector, Kodi switches to the other monitor. if you start the system with the projector being off, Kodi uses the other monitor. now when you turn on the projector, Kodi switches to it because it is configured in gui settings.
running on X11 you can't select a non-connected monitor. if you for example unplug a monitor, X11 updates resolutions and non available monitors wil disappear from this list.